File: wfm-service/app/schemas/flowchart.py
from pydantic import BaseModel, Field


class FlowNode(BaseModel):
    id: str
    label: str
    type: str


class FlowEdge(BaseModel):
    id: str
    source: str
    target: str
    label: str | None = None
    type: str = "DEFAULT"


class Flowchart(BaseModel):
    nodes: list[FlowNode] = Field(default_factory=list)
    edges: list[FlowEdge] = Field(default_factory=list)
    mermaid: str

File: backend/src/main/java/com/reqpilot/model/Flowchart.java
package com.reqpilot.model;

import java.util.List;
import java.util.Objects;

public record Flowchart(List<FlowNode> nodes, List<FlowEdge> edges, String mermaid) {

  public Flowchart {
    nodes = List.copyOf(Objects.requireNonNull(nodes, "nodes are required"));
    edges = List.copyOf(Objects.requireNonNull(edges, "edges are required"));
    Objects.requireNonNull(mermaid, "mermaid is required");
  }
}
